frontend/src/pages/Admin/AddWord/create.tsx

```diff
@@ -0,0 +1,250 @@
+import { Navigate, useParams } from 'react-router-dom';
+import { useFieldArray, useForm } from 'react-hook-form';
+import FormError from '../../../components/FormError';
+import { Key, useEffect } from 'react';
+import axios from 'axios';
+import { config } from '../../../actions/config';
+import { SessionData, Choice, Question } from '../../../actions/types';
+import { connect } from 'react-redux';
+import API from '../../../api/baseAPI';
+
+interface FormData {
+  question: string;
+  choices: {
+    choice: string;
+  }[];
+  is_correct: number;
+  question_image: FileList;
+}
+
+interface Props {
+  currentLogin: SessionData;
+}
+
+const AddWordPage = ({ currentLogin: { user, token } }: Props) => {
+  const questionDataValidation = {
+    question: {
+      required: {
+        value: true,
+        message: 'This field is required',
+      },
+    },
+    choices: {
+      required: {
+        value: true,
+        message: 'Must have at least two choices',
+      },
+      validate: {
+        noDuplicates: (choices: string) =>
+          checkDuplicateFields(choices) || 'No Duplicate Choices',
+      },
+    },
+    is_correct: {
+      required: {
+        value: true,
+        message: 'Must select one correct option',
+      },
+    },
+    question_image: {
+      required: {
+        value: true,
+        message: 'Image is required',
+      },
+      validate: {
+        acceptedTypes: (files: any) =>
+          ['image/png', 'image/gif', 'image/jpeg', undefined].includes(
+            files[0]?.type
+          ) || 'Invalid File Type',
+      },
+    },
+  };
+
+  const { categorySlug } = useParams();
+  const {
+    control,
+    register,
+    handleSubmit,
+    trigger,
+    getValues,
+    formState: { errors },
+  } = useForm<FormData>({
+    defaultValues: {
+      choices: [
+        {
+          choice: '',
+        },
+        {
+          choice: '',
+        },
+      ],
+    },
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'choices',
+    keyName: 'key',
+  });
+
+  const onSubmit = (data: FormData) => {
+    const questionsData = new FormData();
+    questionsData.append('image', data.question_image[0]);
+    questionsData.append('question', data.question);
+
+    const choicesData = data.choices.map(({ choice }, index) => {
+      return {
+        choice,
+        is_correct: index == data.is_correct,
+      };
+    });
+
+    postQuestion(questionsData).then((res) => {
+      postChoices(choicesData, res.id).then((res) => {
+        <Navigate to='/admin/categories' />;
+      });
+    });
```

Comment:
Nested promise works yet if does not offer readability, one of the solution is to use async await,. Nested promise is an [anti-pattern](https://stackoverflow.com/questions/980601/what-is-an-anti-pattern) 
Ref: https://javascript.plainenglish.io/nested-promise-an-anti-pattern-alternative-to-it-a5069923a2ec